### PR TITLE
Correct suspense transactions endpoint input and output schema

### PR DESCRIPTION
**Pull request by adamtry**
Jira Ticket: [Fix the Transactions-API /suspense-account endpoint](https://hackney.atlassian.net/jira/software/projects/HFS/boards/80?selectedIssue=HFS-630)

## What
- Removed redundant boolean `SearchText` parameter, which was functioning as an `IsSuspense` boolean
- Made endpoint return new standardized pagination response
- Created new docker image / service for running tests locally, to closely simulate the pipeline

## Why
- It was agreed that we will not have a use case for querying only non-suspense transactions which made it so that the `SearchText` would always be true. This made it redundant, so it was removed.
- Paginated response was altered so as to be consistent across Finance APIs.
- Old Docker tests container was altered to include SonarCloud analysis. This isn't needed to run tests locally because the scan results won't get sent to the pipeline. It also adds a significant amount of time to the test runs. For this reason we created a new test container without Sonar.

## Notes
- Added a Hackney Shared Finance Nuget package
- Used aliases to resolve the name conflict between the newly added `Hackney.Shared.Finance` Nuget Package, `Hackney.Core.DynamoDb`, and the local copies of these classes that were moved to the `Hackney.Shared.Finance` package.
- Removed `MetadataModel` and `Pagination` (non-package) local classes. These did not have many references and would have been difficult and time consuming to refactor.
- Created parallel versions of `ToResponse` and `ToPaginationDataResponse` in the ResponseFactory. This was done because the alternative would be changing all the namespaces to use the Nuget package. So now the endpoint can use the new namespace while all other functionality remains unaffected with the original namespace.

### CODE REVIEW

**Review by FaisalHackney — APPROVED** [review]

**Comment by Duslerke** [comment]
Merging the PR now to avoid potential future merge conflicts as it got stale after approval.

**Pull request merged**
